File: official/docs/csharp/v6/smartrate/retrieve-standalone-smartrate-deliver-on.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using EasyPost;
using Newtonsoft.Json;

namespace EasyPostExamples
{
    public class Examples
    {
        public static async Task Main()
        {
            var client = new EasyPost.Client(new EasyPost.ClientConfiguration("EASYPOST_API_KEY"));

            EasyPost.Parameters.SmartRate.RecommendShipDateForZipPair recommendShipDateForZipPairParameters = new()
            {
                OriginPostalCode = address1Parameters.Zip,
                DestinationPostalCode = address2Parameters.Zip,
                DesiredDeliveryDate = Fixtures.DesiredDeliveryDate,
                Carriers = ["USPS"],
            };

            EasyPost.Models.API.RecommendShipDateForZipPairResult results = await client.SmartRate.RecommendShipDate(recommendShipDateForZipPairParameters);

            Console.WriteLine(JsonConvert.SerializeObject(results, Formatting.Indented));
        }
    }
}

File: official/docs/ruby/v6/smartrate/retrieve-standalone-smartrate-deliver-on.rb
require 'easypost'

client = EasyPost::Client.new(api_key: 'EASYPOST_API_KEY')

results = client.smart_rate.recommend_ship_date(
  from_zip: '10001',
  to_zip: '10002',
  desired_delivery_date: '2024-07-18',
  carriers: ['ups'],
)

puts results
